lib/Runtime/Library/GlobalObject.cpp

```diff
@@ -2175,27 +2175,28 @@ namespace Js
         return TRUE;
     }
 
-    BOOL GlobalObject::HasItem(uint32 index)
+    PropertyQueryFlags GlobalObject::HasItemQuery(uint32 index)
     {
-        return DynamicObject::HasItem(index)
+        return JavascriptConversion::BooleanToPropertyQueryFlags(JavascriptConversion::PropertyQueryFlagsToBoolean((DynamicObject::HasItemQuery(index)))
```

Comment:
Yes - convert to BOOL for logic operations and back to PropertyQueryFlags to satisfy compiler type-checking.
These long-name converters encapsulates logic that can be updated easily in the future.

---
In reply to: [106305680](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106305680) [](ancestors = 106305680)